src/main/java/org/sonarsource/sonarlint/ls/ServerMain.java

```diff
@@ -19,114 +19,96 @@
  */
 package org.sonarsource.sonarlint.ls;
 
-import java.io.IOException;
 import java.io.PrintStream;
-import java.nio.file.InvalidPathException;
 import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Callable;
 
-public class ServerMain {
+import com.google.common.collect.ImmutableList;
 
-  private static final String ANALYZERS_KEY = "-analyzers";
-  private static final String EXTRA_ANALYZERS_KEY = "-extraAnalyzers";
-  private static final String USAGE = "Usage: java -jar sonarlint-server.jar <jsonRpcPort> " +
-          "[-analyzers path/to/analyzer1.jar [path/to/analyzer2.jar] ...] " +
-          "[-extraAnalyzers path/to/analyzer3.jar [path/to/analyzer4.jar] ...]";
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ParameterException;
+import picocli.CommandLine.Parameters;
+import picocli.CommandLine.Spec;
+import picocli.CommandLine.Model.CommandSpec;
+
+@Command
+public class ServerMain implements Callable<Integer> {
 
   private final PrintStream out;
-  private final PrintStream err;
 
-  public ServerMain(PrintStream out, PrintStream err) {
+  @Parameters(index = "0", description = "The port to which sonarlint should connect to.", defaultValue = "-1")
+  private int deprecatedJsonRpcPort;
+
+  @Option(names = "-jsonRpcPort", description = "The port to which sonarlint should connect to.")
```

Comment:
I don't think the name `jsonRpcPort` is good. Even when using stdio, the protocol is still JSON RPC. Maybe `-socketPort`, or `-clientPort` like in the JDT LS, or simply `-port`?

Reply:
I chose port `-port`.